lib/livebook_web/live/session_live/cell_component.ex

```diff
@@ -238,15 +238,26 @@ defmodule LivebookWeb.SessionLive.CellComponent do
   end
 
   defp cell_body(assigns) do
+    container_class = if length(Access.get(assigns, :comments, [])) >= 1, do: "h-fit", else: ""
+
     ~H"""
     <!-- By setting tabindex we can programmatically focus this element,
          also we actually want to make this element tab-focusable -->
-    <div class="flex relative" data-el-cell-body tabindex="0">
-      <div class="w-1 h-full rounded-lg absolute top-0 -left-3" data-el-cell-focus-indicator>
-      </div>
-      <div class="w-full">
-        <%= render_slot(@inner_block) %>
+    <div class="relative">
+      <div class={"#{container_class} flex-1 flex"}>
+        <div class="flex-1 flex relative mr-3" data-el-cell-body tabindex="0">
+          <div class="w-1 h-full rounded-lg absolute top-0 -left-3" data-el-cell-focus-indicator>
+          </div>
+          <div class="w-full">
+            <%= render_slot(@inner_block) %>
+          </div>
+        </div>
       </div>
+
+      <.live_component module={LivebookWeb.SessionLive.DiscussionComponent}
+        id={"#{@cell_view.id}-discussion"}
```

Comment:
Any reason for using a stateful component here?

Reply:
I wanted to handle Discussion related events in the Discussion component initially. But I saw they're all handled in Session so I followed the same logic.

turned into a stateless component in 0e1a3d4